src/storage/cache/rpc/server.rs

```diff
@@ -0,0 +1,486 @@
+        dispatch_handler: T,
+    ) -> Self {
+        let inner = Arc::new(RpcServerConnectionInner::new(
+            stream,
+            worker_pool,
+            timeout_options,
+            dispatch_handler,
+        ));
+        Self { inner }
+    }
+
+    /// Dispatch the handler for the connection.
+    async fn dispatch(&self, req_header: ReqHeader, done_tx: mpsc::Sender<Vec<u8>>) {
+        // Dispatch the handler for the connection
+        let seq = req_header.seq;
+        let body_len = req_header.len;
+        if let Ok(req_type) = ReqType::from_u8(req_header.op) {
+            debug!(
+                "Dispatch request with header type: {:?}, seq: {:?}, body_len: {:?}",
+                req_type, seq, body_len
+            );
+            if let ReqType::KeepAliveRequest = req_type {
+                // Keep-alive request
+                // Directly send keepalive response to client, do not need to submit to worker pool.
+                // In current implementation, we just send keepalive header to the client stream
+                let resp_header = RespHeader {
+                    seq,
+                    op: RespType::KeepAliveResponse.to_u8(),
+                    len: 0,
+                };
+                // Only one process will access this buffer, so we can use this buffer directly.
+                let resp_buffer: &mut BytesMut = unsafe { &mut *self.inner.req_buf.get() };
+                RespHeader::encode(&resp_header, resp_buffer);
+                if let Ok(res) = self.inner.send_response(resp_buffer).await {
+                    debug!("Sent keepalive response: {:?}", res);
+                } else {
+                    error!("Failed to send keepalive response");
+                }
+            } else {
+                // Try to read the request body
+                match self.inner.recv_len(body_len).await {
+                    Ok(()) => {}
+                    Err(err) => {
+                        error!("Failed to receive request body: {:?}", err);
+                        return;
+                    }
+                };
+                debug!(
+                    "Dispatched handler for the connection, seq: {:?}",
+                    req_header.seq
+                );
+                let req_buffer: &mut BytesMut = unsafe { &mut *self.inner.req_buf.get() };
+                self.inner
+                    .dispatch_handler
+                    .dispatch(req_header, req_buffer, done_tx)
+                    .await;
+            }
+        } else {
+            debug!("Inner request type is not matched: {:?}", req_header.op);
+        }
+    }
+
+    /// Keep the connection and get the handler for the connection.
+    pub async fn run(&self) {
+        // Dispatch the handler for the connection
+        debug!("RpcServerConnection::Received a new TcpStream.");
+
+        // TODO: copy done_tx to the worker pool
+        let (done_tx, mut done_rx) = mpsc::channel::<Vec<u8>>(10000);
```

Comment:
Should use unbound channel to avoid blocking set_result()

Reply:
OK, remove `1000` capacity here.

Reply:
https://docs.rs/tokio/latest/tokio/sync/mpsc/fn.unbounded_channel.html